### DISCUSSION TITLE
Phidget REL1100 with 3583 potentiometer

### DISCUSSION THREAD

**Discussion by bravowren** [artisan-roaster-scope/artisan#588] [Q&A] [answered]
I currently control roaster drum and fan from Artisan using Phidgets REL1100 PWM outputs. Was trying to add an additional rotary control on the roaster itself. 
I think I can log Phidgets 3583 voltages in Artisan (through an analog Vint Hub port), but can these 3583 inputs also control the REL1100 in real time?
Thanks in advance

**Comment by MAKOMO** [comment, marked as answer]
Realtime maybe not, but you can define a quantifier on the logged voltage from the 3583 to turn this channel data into events on changes. Changing the position on the 3583 will then create a new event of the set type, move the corresponding slider and thus change the output of the REL1100PWM. Note that quantifiers are evaluated only once per sampling interval thus you might see a certain delay depending on your sampling interval.

**Reply by bravowren** [reply]
Thank you Makomo for the very helpful response! I will give it a try, and do not think the small delay will be an issue. I like to be hands-on on the roaster, and I will replace the factory rotary potentiometers (currently unused) with 3853s.

**Reply by bravowren** [reply]
Makomo, After defining the quantifiers on the logged voltage from the potentiometers (configured as extra devices - Phidget HUB IO) turning the potentiometer moves the sliders, and logs the event changes just fine. But it does not change the PWM duty cycle output (similar to logging with an an event button without specifying the corresponding PWM action). Any suggestions?
Many thanks for your help.

**Reply by MAKOMO** [reply]
You need to attach a sider action that sends out the PWM duty cycle to the connected Phidget. You need to set the sliders action to "PWM Command" and you a something like "out(0,{})" as action (first number is the output channel and the {} is the placeholder substituted by the event value). Note that all this is well documented in blogs and on the Phidget device page, as well as the slider help page within the app.

**Reply by bravowren** [reply]
Hi Marko, I have indeed set up my slider action to PWM command. Sliders, buttons, and keyboard shortcuts work fine in logging and controlling the PWM output. But Quantifier voltage inputs changes from hub inputs only log event changes and move slider, without changing PWM output. I have not found any place to specify PWM Command action for Quantifiers like there is in Sliders and Buttons. Screen shots are enclosed.
![ExtraDevices](https://user-images.githubusercontent.com/81058869/116967091-6bca6500-ac66-11eb-928d-1fb2df1cc474.png)
![Quantifiers](https://user-images.githubusercontent.com/81058869/116967096-6cfb9200-ac66-11eb-9444-c8f5550ed813.png)
![Buttons](https://user-images.githubusercontent.com/81058869/116967100-6e2cbf00-ac66-11eb-8285-1a576ce53ded.png)
![Slider](https://user-images.githubusercontent.com/81058869/116967102-6ec55580-ac66-11eb-9e5f-7a85511b71e8.png)

**Reply by MAKOMO** [reply]
Well, I forgot totally that event actions triggered by quantifiers do not emit the sliders event action. This is because that mechanism was added to sync the state of a PLC with Artisan where a user could enter changes on its touch screen. In that case the state changes are directly established in the connected PLC and there is not need to send them again from Artisan. Your case is different. I just added another flag to turn slider actions for an event quantifier ON. You will need a new build for this. I'll post you a link to a beta build in short time.

**Reply by bravowren** [reply]
Thank you! And thanks for the tremendous piece of software and the very illuminating documentation both within the App, and also on the blog.

**Reply by MAKOMO** [reply]
Here is the link to a pre-build: [https://www.dropbox.com/s/wt0ozdcra428ias/artisan-mac-2.4.5.dmg?dl=0](https://www.dropbox.com/s/wt0ozdcra428ias/artisan-mac-2.4.5.dmg?dl=0)

**Reply by bravowren** [reply]
Just tried 2.4.5. I checked the new "action" check boxes, and was not able to get the feature to work. PWM output (measured from OUT1100 with a scope) will not track input voltage from potentiometer. Also I came across another issue that is 100% repeatable in both 2.4.4 and 2.4.5. 

Slider and Event logging will track potentiometer ONLY if there is no prior input on keyboard during a session. If I use a button, keyboard command, or slider to change value associated with a quantifier, any following corresponding potentiometer inputs are ignored. 

To clarify, here is a scenario: If I press "Start" and then move the "Air" and "Drum" potentiometers, the corresponding slider values change, and event values are logged just fine. This continues to work only until I change a value from the keyboard. E.g. if I change a value for "Air" from the keyboard, any further inputs from the "Air" potentiometer will be ignored, but values from the the "Drum" potentiometer will continue to track (...until I change a Drum value from the keyboard). 

There is also an intermittent "unhandled exception" error I also came across in the "Quantifiers" dialog box (in both 2.4.4. and 2.4.5) that I will report in the issues section.

**Reply by MAKOMO** [reply]
Bad. I had a tiny typo in my change. Fixed. Now with "active" the slider action should be sent as intended.

https://www.dropbox.com/s/dbfgsggyqu71t36/artisan-win-2.4.5.zip?dl=0

I cannot duplicate the issue regarding button or "keyboard command" preventing further quantification. Could you please check again and file a separate issue if you are still observing this.

Thanks for filing that other issue. Unrelated,  but fixed too.

**Reply by bravowren** [reply]
Marko, is it possible to send me a Mac build?

**Reply by MAKOMO** [reply]
Ah, sorry. Here it is: https://www.dropbox.com/s/ackj79ndya9n1vo/artisan-mac-2.4.5.dmg?dl=0

**Reply by bravowren** [reply]
Thank You! It works great, as expected. I have submitted a separate bug report for the other issue I spoke of. Many thanks.
